src/api/app/jobs/update_released_binaries_job.rb

```diff
@@ -65,6 +65,7 @@ def update_binary_releases_for_repository(repository, new_binary_releases, time
         old_binary_release = old_binary_releases[hashkey_new_binary_releases(backend_binary, backend_binary['medium'])]
         if old_binary_release
           # still exists, do not touch obsolete time
+          old_binary_release = repository.binary_releases.find(old_binary_release[:id])
```

Comment:
I wouldn't mind using a different variable with a different name from this line on, like `old_binary_release_object =....`, or the other way around, using a `old_binary_release_hash` before. It's just tricky to figure out that before this line it was a `hash`, after it is an activerecord object instead. Indeed here it uses the `[:id]` way to access the `id`, the line below uses the `.id` instead on the same variable name.

Reply:
> Indeed here it uses the [:id] way to access the id, the line below uses the .id instead on the same variable name.

Because the content of the variable changes :)

Reply:
Exactly, it is implicitly done, a bit tricky to figure out. But still, it's just fine.